google/cloud/bigtable/internal/operation_context_factory.cc

```diff
@@ -424,6 +463,59 @@ MetricsOperationContextFactory::ReadModifyWriteRow(
       clock_);
 }
 
+std::shared_ptr<OperationContext> MetricsOperationContextFactory::PrepareQuery(
+    std::string const& instance_name, std::string const& app_profile) {
+  auto constexpr kRpc = "PrepareQuery";
+  absl::call_once(prepare_query_metrics_.once, [this, kRpc]() {
+    std::vector<std::shared_ptr<Metric const>> v;
+    v.emplace_back(std::make_shared<OperationLatency>(kRpc, provider_));
+    v.emplace_back(std::make_shared<AttemptLatency>(kRpc, provider_));
+    v.emplace_back(std::make_shared<RetryCount>(kRpc, provider_));
+    v.emplace_back(std::make_shared<ServerLatency>(kRpc, provider_));
+    v.emplace_back(std::make_shared<ConnectivityErrorCount>(kRpc, provider_));
+    swap(prepare_query_metrics_.metrics, v);
+  });
+
+  auto resource_labels = ResourceLabelsFromInstanceName(instance_name);
+  DataLabels data_labels = {kRpc,
+                            "false", /*=streaming*/
+                            "cpp.Bigtable/" + version_string(),
+                            client_uid_,
+                            app_profile,
+                            "" /*=status*/};
+
+  return std::make_shared<OperationContext>(
+      resource_labels, data_labels, prepare_query_metrics_.metrics, clock_);
+}
+
+std::shared_ptr<OperationContext> MetricsOperationContextFactory::ExecuteQuery(
+    std::string const& instance_name, std::string const& app_profile) {
+  auto constexpr kRpc = "ExecuteQuery";
+  absl::call_once(execute_query_metrics_.once, [this, kRpc]() {
+    std::vector<std::shared_ptr<Metric const>> v;
+    v.emplace_back(std::make_shared<OperationLatency>(kRpc, provider_));
+    v.emplace_back(std::make_shared<AttemptLatency>(kRpc, provider_));
+    v.emplace_back(std::make_shared<RetryCount>(kRpc, provider_));
+    v.emplace_back(
```

Comment:
Let's wait on adding ApplicationBlockingLatency. We may need to make some changes to RowStream and RowStreaminIterator to support measuring it.

Reply:
Ok, I removed it for now.